### ISSUE TITLE
Scaling Main Window and Piano Keyboard Widget

### ISSUE THREAD

**Issue by barak** [pianobooster/PianoBooster#199]
When the main window is scaled (e.g., to fill the entire screen) the contents don't really adjust. When using the program at a piano, it seem natural to grow the window, so it would be nice if the additional space were used appropriately.

**Comment by glixx** [comment]
Needs to scale piano keyboard widget too.

**Comment by NicolaasJKotze** [comment]
Adding screenshots as reference to support retesting and regression testing in the future.

**Version**: 0.7.3-PREVIEW1
**Resolution(s)**: 1600x900(_Secondary Display_)   and 1366x768(_Primary Display_)
**Operating System**: Kubuntu 19.10
**KDE Plasma Version**: 5.18.3
**KDE Frameworks Version**: 5.67.0
**Qt Version**: 5.12.4
**Kernel Version**: 5.3.0-46-generic
**OS Type**: 64-bit
**Processors**: 4 × Intel® Core™ i3 CPU M 350 @ 2.27GHz
**Memory**: 7,6 GiB of RAM

**1600x900**
![image](https://user-images.githubusercontent.com/33121021/79783472-00ef0a00-8341-11ea-94fe-8f847bebf789.png)

**1366x768**
![image](https://user-images.githubusercontent.com/33121021/79783232-a0f86380-8340-11ea-97e4-2a612fb3bdd1.png)

**Comment by pianobooster** [comment]
This is probably not worth doing as it. Because on old PC with under powered graphics there is still lot of flickering and video tearing of the horizontal scrolling and that is the reason why the scrolling area is limited. I have spent ages in the past trying to optimise this for low powered machines and it is still a problem on some machines today. Also I really recommend that you don't use the piano keyboard display unless you get totally suck assuming you are trying to use PB to read the music using a piano keyboard.

**Comment by ke7kto** [comment]
> This is probably not worth doing as it.

I love this program, but on a 27" 4K monitor, the staves are tiny. I'm teaching my son how to play (he's loving the program), but his one complaint was, "can you make it bigger? I really need it bigger." Some zoom functionality seems necessary to accomodate multiple screen sizes. I do believe this would be an enhancement.

**Comment by DocSmeal** [comment]
I agree with the other posters here about the need for an option to scale up the staves. Even on a 2019 laptop (3 years old), when in fullscreen mode, the staves do not scale up in size.  I love the program but I hate the squinting to see the notes.  Also, sometimes with chords the notes overlap the sharp and flat signs and a larger scaled up stave would be REALLY helpful.  Even just a manual zoom button would be great.

**Comment by khlsvr** [comment]
+1 same here. My screen where pianobooster is running is a bit further from where my piano is which makes it so that if zooming was possible, I would zoom in in an exaggarated manner.

 In the meanwhile, does anyone know any external magnifier tools that you could use to zoom in the portion of the app, or other software like this that has a zoom in function?
 
 edit:
 
 I tried a few magnifying tools on my Arch linux. 
 
 ```
$  yay -Ss magnify
yay aur/zoomx 0.0.1-1 (+1 0.00) 
    Screen magnifying zoom utility for X11
aur/vmg-svn 3.7-3 (+6 0.01) 
    Handy virtual magnifying glass tool SVN version
aur/vmg 3.5-2 (+7 0.00) 
    Handy virtual magnifying glass tool
aur/imageenlarger 0.9.1-2 (+19 0.09) 
    SmillaEnlarger is a graphical tool to resize, especially magnify bitmaps in high quality
extra/xorg-xmag 1.0.6-3 (20.8 KiB 52.8 KiB) 
    Magnify parts of the screen
$  yay -Ss magnifier
aur/magnifiqus 0.2.1-1 (+0 0.00) 
    Qt based screen magnifier
aur/magnifiqus-git 0.2.0.r0.g93f9847-1 (+0 0.00) 
    Qt based screen magnifier
aur/magnus 1.0.3-3 (+2 0.00) 
    A very simple screen magnifier
aur/picket 1.2-1 (+3 0.04) 
    Linux screen color picker with magnifier and custom user defined formats.
aur/xzoom 0.3-6 (+49 0.05) (Installed)
    A screen magnifier
extra/kmag 21.08.3-1 (792.4 KiB 1.4 MiB) [kde-applications kde-accessibility] 
    Screen Magnifier
```
**zoomx:** It just wanted to take a snapshot of the screen area I clicked, no moving picture.  
**vmg-svn:** This app would have wanted to install dependencies over 1.6G in size so I skipped it. The vmg without the svn version needed just less than 60M so I tried that. I got some gobject errors so I removed the app and didn't start to solve the problem.  
**xorg-xmag:** Same as zoomx, seemed to be just for taking a magnified screenshot of an area.  
**magnifiqus:** After installing this or the -git version, I couldn't get it to work. When I run the magnifiqus executable, it just didn't do anything. It didn't bring me back to the terminal prompt either. I don't know how to debug this problem.  
magnus: This supposedly should do moving frames, but when I placed my mouse over the area it magnifies, it wouldn't update it until I move my mouse again, so it didn't work with pianobooster really.
**xzoom:** This does what I want. I could pick an area and it would update the frames when they moved there. The only problem is that framerate is really bad, something like **2-5 fps**, so it is a little unpleasant to look at, and it could create problems when you have to press keys at faster pace I guess.
**kmag:** This wanted to install a lot of kde dependencies, 220M worth so I skipped this, though I might try it later.

Anyone know if there are other options I should try?

Below is a screenshot of setting up my dual monitor setup so that the pianobooster runs on one screen, and then I lay a zoomed in version with **xzoom** on the other monitor so I can see it from the piano properly. In addition the pianobooster is using an environment variable QT_SCALE_FACTOR=2.0 so that I can see the UI text better.

![2021-12-09_130533](https://user-images.githubusercontent.com/70579677/145387484-4d8fccb6-d6fe-48da-9d41-83715546919d.png)

**Comment by ke7kto** [comment]
I don't know what happened, but the scaling is much better now!!! Thank you!
